examples/detect_onsets_presets.py

```diff
@@ -65,20 +72,31 @@ def process_arguments(args):
     parser.add_argument('output_file',
                         action='store',
                         help='path to the output csv file')
-    parser.add_argument('json_file',
-                        action='store',
-                        help='path to the json file of presets')
     parser.add_argument('preset_name',
                         action='store',
                         help='name of the preset to load (e.g. chico)')
+    parser.add_argument('--json_file',
+                        action='store',
+                        help='path to the json file of presets',
+                        default=os.path.join(dirname,'../carat/presets/onsets.json'),
```

Comment:
Actually I did the dirname = os.path.dirname(__file__) thing so that it's indipendent of where you execute it from, I tried it from a couple of folders and it worked.

true about unix, I'll change it to os.path.join(dirname, os.pardir, 'carat', 'presets', 'onsets.json') which should make it portable

Reply:
Oh, I see. You're right. But I explained myself wrong. Let me clarify.

Think about the following situation. You install the carat library using pip, right? You download the example script to detect the onsets and you place it somewhere in your working path. So you run the script from this path without knowing where the carat library is installed, hence, no idea where the presets file is located. This is an issue, right? What do you think ... OK, not a big deal, I agree :)

This is how I see it. The default presets file should be provided with the library. This is the file that should be loaded by default by the util.load_preset function. Besides, we should have some function for the user to know what presets are available, their parameter values and their intended use. If none of those presets fit the user's needs she could define her own presets.json file and set the parameter values as wanted. 

We don't have to implement all of this right now. But it would be nice if we arrive to a common vision of the best way we should deal with presets in carat. 

Cheers :)

Reply:
Ahh, yes, I see... I'll look into it a bit more